test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -131,20 +146,13 @@ func TestRotatingLogsDoesNotSkipLines(t *testing.T) {
 	awsservice.ValidateLogsInOrder(t, logGroup, logStream, lines, start)
 }
 
-func writeLogs(t *testing.T, filePath string, iterations int) {
-	f, err := os.Create(filePath)
-	if err != nil {
-		t.Fatalf("Error occurred creating log file for writing: %v", err)
-	}
-	defer f.Close()
-	defer os.Remove(filePath)
```

Comment:
writeFile() was deleting the file.
If it is less than 1 second between file creation and deletion then the CWA probably does not discover the file.
Also If you delete the file <200ms after the last line is written then there is a chance CWA does not read the last line.